### ISSUE TITLE
Can't use in non-typescript codebases, all files seems to be .tx only

### ISSUE THREAD

**Issue by ziaulrehman40** [wcandillon/react-native-responsive-ui#59]
I am trying to use this library in a project which is not using typescript.

I am getting error `Module not found: Can't resolve 'react-native-responsive-ui' in.....` on closer inspection it seems all files in both `dist` and `src` are `ts` or `tsx`, and readme doesn't indicate anything about this typescript transpilation dependency.

Normally, npm packages are released with compiled ES5 or ES6 code. Which make them usable universally.

Version installed: `2.1.1`

<img width="304" alt="image" src="https://user-images.githubusercontent.com/22441482/62272188-5a985880-b3ef-11e9-8bcb-77dd44e4d9e5.png">

**UPDATE:**
Part of the problem was the absence of `index.js` on the root. I don't know if i am doing something wrong, or there is some-other way of suing this and similar architecture packages.

What i have done for now, is to [clone to generate js files](https://github.com/ziaulrehman40/react-native-responsive-ui) and import like this: `import { useDimensions } from 'react-native-responsive-ui/dist/module';`

**Comment by nabilfreeman** [comment]
Agreed - also this results in a really annoying eslint error even if you have transpilation because `package.json` points to `index.js` rather than `index.ts`.

I fixed with `patch-package`:

To include `package.json` which is normally excluded I needed to run

`npx patch-package --exclude 'nothing' react-native-responsive-ui`

after making the edit.

```
diff --git a/node_modules/react-native-responsive-ui/package.json b/node_modules/react-native-responsive-ui/package.json
index a2588ab..37af7b1 100644
--- a/node_modules/react-native-responsive-ui/package.json
+++ b/node_modules/react-native-responsive-ui/package.json
@@ -6,28 +6,34 @@
       "typescript"
     ]
   },
  "_from": "https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz",
   "_id": "react-native-responsive-ui@2.1.1",
   "_inBundle": false,
   "_integrity": "sha512-9NUHI4oPKHWQTg6Ls8QQhR38/dgWUFkixPeCtzvcNtcutDClMAbpI47WoOKXaet2rgxdyojPexsIRK67x18RYQ==",
   "_location": "/react-native-responsive-ui",
   "_phantomChildren": {},
   "_requested": {
     "type": "remote",
     "raw": "react-native-responsive-ui@https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz",
     "name": "react-native-responsive-ui",
     "escapedName": "react-native-responsive-ui",
     "rawSpec": "https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz",
     "saveSpec": "https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz",
     "fetchSpec": "https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz"
   },
   "_requiredBy": [
     "/"
   ],
   "_resolved": "https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz",
   "_shasum": "d46557bae2e1974d5d768936091abd483587c63d",
   "_spec": "react-native-responsive-ui@https://registry.npmjs.org/react-native-responsive-ui/-/react-native-responsive-ui-2.1.1.tgz",
   "_where": "/private/var/folders/ws/3scc52zj0k5bbkg072m1bfsr0000gp/T/tmp-37033YK93zOBbQ3sP",
   "author": {
     "name": "William Candillon",
     "email": "wcandillon@gmail.com"
@@ -35,11 +41,9 @@
   "bugs": {
     "url": "https://github.com/wcandillon/react-native-responsive-ui/issues"
   },
   "bundleDependencies": false,
   "dependencies": {
     "lodash": "^4.17.11"
   },
   "deprecated": false,
   "description": "Responsive UIs for React Native",
   "devDependencies": {
     "@react-native-community/bob": "^0.4.1",
@@ -55,14 +59,14 @@
   ],
   "homepage": "https://github.com/wcandillon/react-native-responsive-ui#readme",
   "license": "MIT",
-  "main": "dist/src/index.js",
+  "main": "dist/src/index.ts",
   "name": "react-native-responsive-ui",
   "peerDependencies": {
     "react": ">=16.8.3",
     "react-native": "*"
   },
   "private": false,
   "react-native": "src/index.ts",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/wcandillon/react-native-responsive-ui.git"
```

**Comment by nabilfreeman** [comment]
I'm back again! This now caused crashes in my react-native-web migration so I had to add support for Typescript to webpack. Looks like RN Metro Bundler supports TS out of the box.

**Comment by crutchcorn** [comment]
This issue should now be fixed in `2.1.2`. Please mention if this issue persists in future projects

**Comment by nabilfreeman** [comment]
Thank you @crutchcorn !!! We are trying the new version as part of our pre xmas RN upgrade
